### PR TITLE
Feature/server statistics

### PR DESCRIPTION
**Pull request by amukhsimov**

### CODE REVIEW

File: components/restreamer/src/server.rs [hunk 1]
```diff
@@ -729,6 +730,141 @@ pub mod callback {
...
+
+    /// Runs statistics monitoring
+    /// # Panics
+    ///
+    /// # Errors
+    /// No errors expected.
+    #[allow(clippy::cast_possible_truncation)]
+    #[allow(clippy::cast_precision_loss)]
+    pub async fn run(state: State) -> Result<(), Failure> {
+        // we use tx_last and rx_last to compute the delta
```
Comment (skhalymon) [inline components/restreamer/src/server.rs:752]: comment duplicate

File: components/restreamer/src/server.rs [hunk 2]
```diff
@@ -729,6 +730,141 @@ pub mod callback {
...
+
+                    // Update cpu usage
+                    match sys.cpu_load_aggregate() {
+                        Ok(cpu) => {
+                            // Need to wait some time to let the library compute
+                            // CPU usage.
+                            // Do not change delay time, since it is also used
+                            // further to compute network statistics
+                            // (bytes sent/received last second)
+                            time::delay_for(Duration::from_secs(1)).await;
```
Comment (skhalymon) [inline components/restreamer/src/server.rs:775]: If delay time used in different places then maybe better to extract it to separate variable?

File: components/restreamer/src/server.rs [hunk 3]
```diff
@@ -729,6 +730,141 @@ pub mod callback {
...
+    use tokio::time;
+
+    use crate::{cli::Failure, display_panic, state::ServerInfo, State};
+    use ephyr_log::log;
+    use futures::FutureExt;
+    use std::panic::AssertUnwindSafe;
+
+    /// Runs statistics monitoring
+    /// # Panics
+    ///
```
Comment (skhalymon) [inline components/restreamer/src/server.rs:746]: Maybe add smth: "Panic is captured to log. Could be panicked during getting server statistics"

File: components/restreamer/src/server.rs [hunk 4]
```diff
@@ -729,6 +730,141 @@ pub mod callback {
...
+    /// # Errors
+    /// No errors expected.
```
Comment (skhalymon) [inline components/restreamer/src/server.rs:748]: "Errors are captured to log. Could be failed during getting CPU, Memory or Network statistic"

File: components/restreamer/src/state.rs [hunk 5]
```diff
@@ -96,6 +96,62 @@ impl Default for Settings {
...
+    pub fn update_ram(
+        &mut self,
+        ram_total: Option<f64>,
+        ram_free: Option<f64>,
+    ) {
+        self.ram_total = ram_total;
+        self.ram_free = ram_free;
+    }
+
+    /// Updates traffic usage info of this [`ServerInfo`], in megabites
```
Comment (skhalymon) [inline components/restreamer/src/state.rs:144]: megabites -> megabytes

File: components/restreamer/src/server.rs (hunk 2 above)
Comment (PavelKushnirchuk) [reply components/restreamer/src/server.rs:775]: No. It's used only once here

File: components/restreamer/src/server.rs (hunk 4 above)
Comment (PavelKushnirchuk) [reply components/restreamer/src/server.rs]: Extended `Errors` section with comment

**Review by skhalymon — APPROVED** [review]

**Pull request merged**